### PR TITLE
feature: remember checkboxes in script view

### LINKED ISSUE #9177: Feature request: In IDE script table view remember checkbox choices
When making changes to a script, the "show archived" and "show detailed disclosure" choices get reset. Would be more convenient if these keep my last setting

### PR DESCRIPTION
**Pull request by robin-da**
Fixes #9177. This makes the script view remember the choices for
'show_archive', 'show detailed disclosure' and 'table/transaction' view
accross changes to the script.

Not sure if there is a test suite availabe for the extension where tests for
this could be added.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/daml-extension/src/webview.js [hunk 1]
```diff
@@ -34,9 +44,28 @@ window.addEventListener('message', event => {
...
+      if (message.value.showDetailedDisclosure) {
+        document.body.classList.remove('hidden_disclosure');
+        document.body.classList.add('show_disclosure');
+        document.getElementById('show_detailed_disclosure').checked = true;
+      } else {
+        document.body.classList.remove('show_disclosure');
+        document.body.classList.add('hidden_disclosure');
+        document.getElementById('show_detailed_disclosure').checked = false;
+      }
+
```
Comment (cocreature) [inline compiler/daml-extension/src/webview.js:68]: I think you can factor this out into a function that accepts 4 arguments:
1. The boolean that controls the state.
2. The class that is active when the boolean is true.
3. The class that is active when the boolean is false
4. The id of the checkbox.

Comment (cocreature) [reply compiler/daml-extension/src/webview.js:68]: You could also try fitting the transaction vs table view in there. Not sure if that’s helpful or not.

File: compiler/daml-extension/src/webview.js [hunk 2]
```diff
@@ -3,17 +3,27 @@
...
+  });
 }
 function toggle_detailed_disclosure() {
-  document.body.classList.toggle('hidden_disclosure', !document.getElementById('show_detailed_disclosure').checked);
+  const isChecked = document.getElementById('show_detailed_disclosure').checked;
+  document.body.classList.toggle('hidden_disclosure', !isChecked);
+  vscode.postMessage({
+    'command': 'set_show_detailed_disclosure',
+    'value': isChecked
+  });
```
Comment (cocreature) [inline compiler/daml-extension/src/webview.js:19]: These two can also be factored into one.

**Review by cocreature — APPROVED** [review]
Thanks! We don’t have tests for this atm so I’m fine with omitting them. A few suggestions for cleanup.

File: compiler/daml-extension/src/webview.js (hunk 1 above)
Comment (robin-da) [reply compiler/daml-extension/src/webview.js]: factored out in a function.

File: compiler/daml-extension/src/webview.js (hunk 2 above)
Comment (robin-da) [reply compiler/daml-extension/src/webview.js:19]: factored out.

**Pull request merged**